pkg/maintainer/wallet/chain.go

```diff
@@ -0,0 +1,13 @@
+package wallet
+
+import (
+	"github.com/keep-network/keep-core/pkg/tbtc"
+)
+
+// Chain represents the interface that the wallet maintainer module expects to interact
+// with the anchoring blockchain on.
+type Chain interface {
+	// TODO: Change to something more specific once https://github.com/keep-network/keep-core/issues/3632
+	// is handled.
```

Comment:
Do we need to wait for #3632? It's a new package so we should define its own `Chain interface`. It does not matter what the `coordinator` package does - it should be handled in #3632.

The Ethereum implementation of tBTC chain will have to provide all the functions needed but we can limit the interface scope to this package needs only.

Reply:
Unfortunately, we need to pass the chain handle to `coordinator.ProposeDepositsSweep` which passes it to `tbtc.ValidateDepositSweepProposal`.

Reply:
I improved it in 0041f9bb3d70d65d505356cd28798f6384b5ab90 so the Chain interface in this package should be fine now.